src/services/solana-client/SolanaClient.ts

```diff
@@ -0,0 +1,137 @@
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
+import { Connection, PublicKey } from '@solana/web3.js'
+
+import { solanaNFTToCollectible } from 'containers/collectibles/solCollectibleHelpers'
+import { CollectibleState } from 'containers/collectibles/types'
+
+import { SolanaNFT } from './types'
+
+const SOLANA_CLUSTER_ENDPOINT = process.env.REACT_APP_SOLANA_CLUSTER_ENDPOINT
+
+const METADATA_PROGRAM_ID = new PublicKey(
+  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s'
+)
+
+class SolanaClient {
+  private connection = new Connection(SOLANA_CLUSTER_ENDPOINT!, 'confirmed')
+
+  async getAllCollectibles(wallets: string[]): Promise<CollectibleState> {
+    const tokenAccountsByOwnerAddress = await Promise.all(
+      wallets.map(async address =>
+        client.connection.getParsedTokenAccountsByOwner(
+          new PublicKey(address),
+          {
+            programId: TOKEN_PROGRAM_ID
+          }
+        )
+      )
+    )
+
+    const likelyNFTsByOwnerAddress = tokenAccountsByOwnerAddress
+      .map(ta => ta.value)
+      // value is an array of parsed token info
+      .map((value, i) => {
+        const ownerAddress = wallets[i]
+        const mintAddresses = value
+          .map(v => ({
+            mint: v.account.data.parsed.info.mint,
+            tokenAmount: v.account.data.parsed.info.tokenAmount
+          }))
+          .filter(({ tokenAmount }) => {
+            // nfts generally have a supply of 1 and they have 0 decimal places
+            // return tokenAmount.uiAmount === 1 && tokenAmount.decimals === 0
```

Comment:
This looks like a TODO

Reply:
that's right, need to decide to do or not to do